sdcm/reporting/tooling_reporter.py

```diff
@@ -85,3 +96,20 @@ def _collect_version_info(self) -> None:
         self.version = f"{result.get('cassandra-stress', '#FAILED_CHECK_LOGS')}"
         if driver_version := result.get("scylla-java-driver"):
             self.version += f" (with java-driver {driver_version})"
+            self.driver_version = driver_version
+
+        cached_driver_versions: list[str] = self.VERSION_CACHE.get(self.version, [])
+        if self.driver_version:
+            cached_driver_versions.append(self.driver_version)
+        self.VERSION_CACHE[self.version] = cached_driver_versions
```

Comment:
IIUC  we populate cache and then we verify if it's inside the cache - won't `_is_duplicate_version` always return True?

Reply:
I've discussed it with @k0machi offline, and we should be trying to use `@cache` instead to make this work as expected

Reply:
> IIUC we populate cache and then we verify if it's inside the cache - won't `_is_duplicate_version` always return True?

the idea is to always return True when there's a version in the cache but no driver - meaning we are using cassandra stress that doesn't report its driver version, but I'm planning to rework this a bit into using lru_cache instead on report method, so it's going to be work a bit differently anyway